File: Linguagem C Exercicios/Vetores/ALTURAS.c
#include <stdio.h>
#include <stdlib.h>

int main()
{

    int n,cont;
    double alturaMedia,soma,porcentagem;

    printf("Quantas pessoas serao digitadas? ");
    scanf("%d",&n);

    char nomes[n][50];
    int idades[n];
    double alturas[n];


    for (int i=0;i<n;i++){
        printf("Dados da %da pessoa:\n",i+1);
        printf("Nome: ");
        fseek(stdin,0,SEEK_END);
        gets(nomes[i]);
        printf("Idades: ");
        scanf("%d",&idades[i]);
        printf("Altura: ");
        scanf("%lf",&alturas[i]);
        printf("\n");
    };

    soma = 0;
    alturaMedia = 0;

    for (int i = 0;i<n;i++){
        soma = soma + alturas[i];
    };
    alturaMedia = soma / n;
    printf("Altura Media: %.2lf\n",alturaMedia);

    cont =0;

    for (int i =0;i<n;i++){
        if (idades[i] < 16){
            cont = cont + 1;
        };
    };

    porcentagem = (cont * 100.0) / n;
    printf("Percentual de menores de 16 anos: %.1lf %%\n",porcentagem);

    for (int i =0;i<n;i++){
        if (idades[i] < 16){
            printf("%s\n",nomes[i]);
        };
    };


    return 0;
}

File: Linguagem Java/Exercicios/Alturas.java
import java.util.Locale;
import java.util.Scanner;

public class Alturas {

	public static void main(String[] args) {
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int n,cont;
		double soma,media,porcentagem;
		
		System.out.print("Quantas pessoas serao digitadas? ");
        n = sc.nextInt();
        
        int idade[] = new int[n];
        double altura[] = new double[n];
        String nome[] = new String[n];
        
        // Lendo o vetor
        for (int i=0;i<n;i++) {
        	System.out.println("Dados da "+(i+1)+"a pessoa:");
        	System.out.print("Nome: ");
        	sc.nextLine();
        	nome[i] = sc.nextLine();
        	System.out.print("Idade: ");
        	idade[i] = sc.nextInt();
        	System.out.print("Altura: ");
        	altura[i] = sc.nextDouble();        	
        };
        
        System.out.println();
        // Altura media
        soma = 0;
        for (int i=0;i<n;i++) {
        	soma = soma + altura[i];        	
        };
        media = soma / n;
        System.out.println("Altura media: "+String.format("%.2f", media));
        
        // Pessoas com menos de 16 anos
        cont = 0;
        for (int i=0;i<n;i++) {
        	if (idade[i] < 16) {
        		cont++;       		
        	};        	
        };
        
        porcentagem = (cont * 100) / n;
        System.out.println("Pessoas com menos de 16 anos: "+String.format("%.1f", porcentagem)+"%");
        
        for (int i=0;i<n;i++) {
        	if (idade[i] < 16) {
        		System.out.println(nome[i]);      		
        	};        	
        };
        
        sc.close();
	}

}
